tests/manual-test-cases/Group5-Functional-Tests/5-10-Multiple-Datacenter.robot

```diff
@@ -17,11 +17,40 @@ Documentation  Test 5-10 - Multiple Datacenters
 Resource  ../../resources/Util.robot
 Suite Teardown  Run Keyword And Ignore Error  Nimbus Cleanup  ${list}
 
+*** Keywords ***
+# Insert elements from dict2 into dict1, overwriting conflicts in dict1 & returning new dict
+Combine Dictionaries
+    [Arguments]  ${dict1}  ${dict2}
+    ${dict2keys}=  Get Dictionary Keys  ${dict2}
+    :FOR  ${key}  IN  @{dict2keys}
+    \    ${elem}=  Get From Dictionary  ${dict2}  ${key}
+    \    Set To Dictionary  ${dict1}  ${key}  ${elem}
+    [Return]  ${dict1}
+
 *** Test Cases ***
 Test
     Log To Console  \nStarting test...
-    ${esx1}  ${esx1-ip}=  Deploy Nimbus ESXi Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
-    ${esx2}  ${esx2-ip}=  Deploy Nimbus ESXi Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    &{esxes}=  Create Dictionary
+    ${num_of_esxes}=  Evaluate  2
+    :FOR  ${i}  IN RANGE  3
+    # Deploy some ESXi instances
+    \    &{new_esxes}=  Deploy Multiple Nimbus ESXi Servers in Parallel  ${num_of_esxes}  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}
+    \    ${esxes}=  Combine Dictionaries  ${esxes}  ${new_esxes}
+
+    # Investigate to see how many were actually deployed
+    \    ${len}=  Get Length  ${esxes}
+    \    ${num_of_esxes}=  Evaluate  ${num_of_esxes} - ${len}
+
+    # Exit if we've got enough & continue loop if we don't
+    \    Exit For Loop If  ${len} >= ${num_of_esxes}
```

Comment:
actually, I think Exit for loop if ${num_of_esxes} = 0 is what we really want.

+    ${num_of_esxes}=  Evaluate  2
Is outside of the for loop, so it never gets reset to 2.